app/models/validations/date_validations.rb

```diff
@@ -59,6 +59,16 @@ def validate_startdate(record)
     if record["mrcdate"].present? && record.startdate < record["mrcdate"]
       record.errors.add :startdate, I18n.t("validations.setup.startdate.after_major_repair_date")
     end
+
+    location_inactive_status = inactive_status(record.startdate, record.location&.location_deactivation_periods, record.location&.available_from)
```

Comment:
This is for my own learning benefit, sorry! Is `record.location&.location_deactivation_periods` a shorthand for getting all rows (if any exist) from the `location_deactivation_periods` table for the location in question?

Reply:
`record.location&.location_deactivation_periods`
is the same as writing
`record.location.location_deactivation_periods if record.location.present?` I think